api/src/OpenTelemetry/Internal/Logging/Types.hs

```diff
@@ -43,8 +53,76 @@ data Logger = Logger
 
 {- | This is a data type that can represent logs from various sources: application log files, machine generated events, system logs, etc. [Specification outlined here.](https://opentelemetry.io/docs/specs/otel/logs/data-model/)
 Existing log formats can be unambiguously mapped to this data type. Reverse mapping from this data type is also possible to the extent that the target log format has equivalent capabilities.
+Uses an IORef under the hood to allow mutability.
+-}
+data LogRecord a = LogRecord Logger (IORef (ImmutableLogRecord a))
```

Comment:
Stores a `Logger` for access to `InstrumentationScope` in the `Logger`, `AttributeLimits` in the `LoggerProvider`, and `Resource` in the `LoggerProvider`.